potassco/reify.h

```diff
@@ -0,0 +1,150 @@
+// {{{ MIT License
+
+// Copyright 2017 Roland Kaminski
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to
+// deal in the Software without restriction, including without limitation the
+// rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+// sell copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+
+// }}}
+
+#pragma once
+
+#include <cstdint>
+#include <potassco/aspif.h>
+#include <potassco/graph.h>
+#include <potassco/reify_utils.h>
+
+#include <unordered_map>
+#include <vector>
+
+namespace Potassco {
+//! Writes a program in reified facts format to the given output stream.
+/*!
+ * \ingroup WriteType
+ */
+class Reifier : public Potassco::AbstractProgram {
+public:
+    //! Creates a new object and associates it with the given output stream.
+    /*!
+     * \param out Output stream to which program is written.
+     * \param calculateSCCs Whether to compute strongly connected components.
+     * \param reifyStep Whether to include step information in output.
+     */
+    explicit Reifier(std::ostream& out, bool calculateSCCs, bool reifyStep);
```

Comment:
Nitpick style: Maybe consider replacing the two consecutive bools with a flag set or strong types. This make call sites clearer and prevents accidental mixups.